src/common/io/src/geometry.rs

```diff
@@ -19,22 +19,25 @@ use geozero::CoordDimensions;
 use geozero::ToWkb;
 use wkt::TryFromWkt;
 
-pub fn parse_to_ewkb(buf: &[u8]) -> Result<Vec<u8>> {
+pub fn parse_to_ewkb(buf: &[u8], srid: Option<i32>) -> Result<Vec<u8>> {
     let wkt = std::str::from_utf8(buf).map_err(|e| ErrorCode::GeometryError(e.to_string()))?;
-    let mut srid: Option<i32> = None;
     let input_wkt = wkt.trim().to_ascii_uppercase();
 
     let parts: Vec<&str> = input_wkt.split(';').collect();
 
-    if input_wkt.starts_with("SRID=") && parts.len() == 2 {
-        srid = Some(parts[0].replace("SRID=", "").parse()?);
-    }
+    let parsed_srid: Option<i32> = srid.or_else(|| {
+        if input_wkt.starts_with("SRID=") && parts.len() == 2 {
+            parts[0].replace("SRID=", "").parse().ok()
+        } else {
+            None
```

Comment:
[links](https://docs.snowflake.com/en/sql-reference/functions/st_geometryfromwkt)
Snowflake default processing for WKT input. if the srid argument is not specified, the resulting GEOMETRY object has the SRID set to 0.

Reply:
0 is not a valid SRID

Reply:
PostGIS as a factual standard in gis system. We can refer to its [settings](https://postgis.net/docs/manual-3.5/using_postgis_dbmanagement.html#spatial_ref_sys). A local SRS is a Cartesian coordinate system which is not referenced to the earth's surface. In PostGIS this is specified by a SRID value of 0. And all geo rust library and posgis  depend on [libgeos](https://libgeos.org/doxygen/geos__c_8h.html). GEOSGetSRID function defines srid 0 is unknown / not set.